### ISSUE TITLE
Fix gravity when exiting ladders

### ISSUE THREAD

**Issue by satoon101** [MarkusMeskanen/EasyPlayer#4]
A player's gravity is reset when entering a ladder.  It would be great if EasyPlayer handled resetting the gravity back once the player exits the ladder.

**Comment by Mahi** [comment]
Good idea, this should definitely be something `EasyPlayer` handles! I'm in a hurry right now but I'll investigate this tomorrow or next weekend. Are you aware of an easy way to listen to a player exiting a ladder?

**Comment by satoon101** [comment]
I didn't do any of the research or write any of the code, but GunGame51 has an implementation that can be found here:

https://github.com/GunGame-Dev-Team/GunGame51/blob/master/cstrike/addons/eventscripts/gungame51/scripts/included/gg_multi_level/gg_multi_level.py

**Comment by satoon101** [comment]
There might be dynamic functions when entering/exiting, but I have not yet looked into that.

**Comment by satoon101** [comment]
Finally had some time to do some research.  I tried hooking `CBasePlayer::ExitLadder`, but it spammed the console and crashed the server as I was connecting with no bots or other players...

I cannot figure out the signature of `CFuncLadder::PlayerGotOn` and `CFuncLadder::PlayerGotOff`.  If someone could do that, it would be ideal (after testing, of course).  Though, those would need to be added to our data for each engine/game that has those 2 functions.

Outside of that, it seems that checking the player's `movetype` and `flags` with a tick listener might have to suffice, unfortunately.
